### DISCUSSION TITLE
How to set up PWM frequency

### DISCUSSION THREAD

**Discussion by DS-567** [stnolting/neorv32#1054] [General]
I can't figure out how to set a PWM frequency of any value between 10kHz and 20kHz using the `neorv32_pwm_setup()` function.

Note I am using an old version of Neorv32 that only accepts one parameter as input to this function, i.e. the clock prescaler 3-bits.

I can achieve 6.1kHz with the prescaler_64 and the next prescaler jumps to 48kHz. How can I achieve programmable PWM frequency?

**Comment by stnolting** [comment]
With the old version (before v1.10.5.4 / #1049), the PWM frequency could only be configured via the 3-bit clock divider - which only allowed a very limited spectrum:

https://github.com/stnolting/neorv32/blob/a9a8e20ca38059b7d0a091d1070c9f286d7db9c2/sw/lib/include/neorv32.h#L210-L219

This is precisely why the PWM module has been revised and now offers far more options for setting the PWM frequency.

**Reply by DS-567** [reply]
Ah right that makes sense, thanks Stephan. 

Any hacks or workarounds? 😅

**Reply by stnolting** [reply]
Oh, that's complicated... The easiest “hack” would be to upgrade to the latest version. 😅

**Reply by DS-567** [reply]
Haha, I might have to. Thanks anyway Stephan 👍
